### PR TITLE
Crewkin Abilities v1

### PR DESCRIPTION
**Pull request by MediHound**
## About The Pull Request

Adds two of the Shadekin abilities for use by Black-Eyed Shadekin.

## Why It's Good For The Game

Species Abilities are fun and are good! They are what turn "I'm you but slower and with 25 more HP" into "I'm a different species."
We already have some species getting some abilities, and we also have some species getting abilities that are a nod to their parent species.

Black-Eyed Shadekin are Black-Eyed because in lore they weren't able to expand their Energy Reservoir far enough during The Ritual to allow for Plane Shifting, or because they have damaged their organ which is responsible for gathering Dark Energy, But this being an "All or Nothing" is a bit weird. Sure, some Black-Eyed Shadekin have completely lost their ability to interact with Dark Energy, but some have possibly damaged their ability only a bit, or weren't able to fully expand their ability.

With this PR, Black-Eyed Shadekin get an energy pool of 50, half of what Shadekin usually have. Along with that they gain access to the "Create Shade" and "Regenerate Other" abilities.
Those abilities are nice things that replicate items (Photon-Disruption Grenade, Inaprovaline Injector), which aren't inherently game-breaking, but will give Black-Eyed Shadekin something unusual which turns them from "Basically an anthro canine" to "Oh yeah, that species is different!"

The intention here is to give Black-Eyed Shadekin something unusual that is marginally useful in some situations but not game-breaking.
Create Shade allows for shenanigans, but is basically the same as using a Photon Disruption grenade from Maintenance.
Regenerate Other allows for minor healing of other people, using a medikit is faster and unlike this ability, a medikit can be used on yourself.

ToDo for later versions (help greatly appreciated):
Make two traits which set the energy maximum to 25 or 0, so people can pick how much of the abilities they have access to. (Two brains for this purpose have been made, needs a Trait which changes the brain the Crewkin spawns with.)

## Changelog
:cl:
add: Black-Eyed Shadekin get access to powers
add: Black-Eyed Shadekin powers, which are weaker versions of the Shadekin ones, minus the teleport
tweak: Species ID and Species UID for Shadekin and Black-Eyed Shadekin
/:cl:

### CODE REVIEW

**Comment by MediHound** [comment]
Conflict with #4610 
Both modify "_onclick/HUD/_defines.dm" and "citadel.dme"
After #4610 is merged, this PR needs to change file "code/_onclick/hud/_defines.dm", line 168, changing "East" to "Right".

**Comment by MediHound** [comment]
Nevermind, did the necessary change now. Once #4610 is merged, this is ready for testmerge.

**Comment by MediHound** [comment]
Merge conflict should be resolved now.

File: code/modules/species/shadekin/crew_shadekin_abilities.dm [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+	ability_icon_state = "tech_biomedaura"
+
+/mob/living/carbon/human/proc/crewkin_regenerate_other()
+	set name = "Regenerate Other (50)"
+	set desc = "Spend energy to heal physical wounds in another creature."
+	set category = "Shadekin"
+
+	var/ability_cost = 50
+
+	if(species.get_species_id() != SPECIES_ID_SHADEKIN)
```
Comment (TheLordME) [inline code/modules/species/shadekin/crew_shadekin_abilities.dm:24]: Shouldnt this be SPECIES_ID_SHADEKIN_BLACK_EYED?

File: code/modules/species/shadekin/crew_shadekin_abilities.dm [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+	ability_icon_state = "tech_dispelold"
+
+/mob/living/carbon/human/proc/crewkin_create_shade()
+	set name = "Create Shade (25)"
+	set desc = "Create a field of darkness that follows you."
+	set category = "Shadekin"
+
+	var/ability_cost = 25
+
+	if(species.get_species_id() != SPECIES_ID_SHADEKIN)
```
Comment (TheLordME) [inline code/modules/species/shadekin/crew_shadekin_abilities.dm:88]: same as above, shouldnt this be
SPECIES_ID_SHADEKIN_BLACK_EYED
?

File: code/__DEFINES/species/species.dm [hunk 3]
```diff
@@ -7,6 +7,8 @@
 #define SPECIES_ID_PROMETHEAN "promethean"
 #define SPECIES_ID_TESHARI "teshari"
 #define SPECIES_ID_VOX "vox"
+#define SPECIES_ID_SHADEKIN "shadekin"
```
Comment (TheLordME) [inline code/__DEFINES/species/species.dm:10]: why define an additional ID, shadekin and crewkin are already distinct as species names

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 4]
```diff
@@ -419,42 +419,96 @@
...
 	if(!istype(SK))
-		return FALSE
+		if(!istype(BESK))
+			return FALSE
 
 	if(amount > 0 || !(SK.check_infinite_energy(src)))
 		var/new_amount = SK.get_energy(src) + amount
 		SK.set_energy(src, new_amount)
+
+/*
```
Comment (TheLordME) [inline code/modules/mob/living/carbon/human/human_powers.dm:472]: you are handleing it in the lines above, 
get rid of the unessessary commented out code please
(its acceptable to leave commented code out, if it fullfills a different function that the new code you are replacing it with

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 5]
```diff
@@ -419,42 +419,96 @@
 
 /mob/living/carbon/human/proc/shadekin_get_energy()
 	var/datum/species/shadekin/SK = species
+	var/datum/species/crew_shadekin/BESK = species
 
 	if(!istype(SK))
```
Comment (TheLordME) [inline code/modules/mob/living/carbon/human/human_powers.dm:424]: wont work like that
this should work
```suggestion
	if(istype(SK))
		return SK.get_energy()
	if(istype(BESK))
		return BESK.get_energy()
	return false
```

and please dont use all caps variables. Defines are all caps, variables always start lower case
I know its not your doing but please adjust it while you are at it.

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 6]
```diff
@@ -419,42 +419,96 @@
...
+			return FALSE
 
 	return SK.get_energy(src)
 
 /mob/living/carbon/human/proc/shadekin_get_max_energy()
 	var/datum/species/shadekin/SK = species
+	var/datum/species/crew_shadekin/BESK = species
 
 	if(!istype(SK))
-		return FALSE
```
Comment (TheLordME) [inline code/modules/mob/living/carbon/human/human_powers.dm:432]: see above

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 7]
```diff
@@ -419,42 +419,96 @@
...
 
 	return SK.get_max_energy(src)
 
 /mob/living/carbon/human/proc/shadekin_set_energy(new_energy)
 	var/datum/species/shadekin/SK = species
+	var/datum/species/crew_shadekin/BESK = species
 
 	if(!istype(SK))
-		return FALSE
+		if(!istype(BESK))
```
Comment (TheLordME) [inline code/modules/mob/living/carbon/human/human_powers.dm:445]: see above

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 8]
```diff
@@ -419,42 +419,96 @@
...
 	SK.set_energy(src, new_energy)
 
 /mob/living/carbon/human/proc/shadekin_set_max_energy(new_max_energy)
 	var/datum/species/shadekin/SK = species
+	var/datum/species/crew_shadekin/BESK = species
 
 	if(!istype(SK))
-		return FALSE
+		if(!istype(BESK))
+			return FALSE
```
Comment (TheLordME) [inline code/modules/mob/living/carbon/human/human_powers.dm:456]: see above

File: code/modules/mob/living/carbon/human/human_powers.dm [hunk 9]
```diff
@@ -419,42 +419,96 @@
...
 	SK.set_max_energy(src, new_max_energy)
 
 /mob/living/carbon/human/proc/shadekin_adjust_energy(amount)
 	var/datum/species/shadekin/SK = species
+	var/datum/species/crew_shadekin/BESK = species
 
 	if(!istype(SK))
-		return FALSE
+		if(!istype(BESK))
+			return FALSE
```
Comment (TheLordME) [inline code/modules/mob/living/carbon/human/human_powers.dm:466]: see above

File: code/modules/mob/mob_defines.dm [hunk 10]
```diff
@@ -343,6 +343,7 @@
 	var/disconnect_time = null
 
 	var/atom/movable/screen/shadekin/shadekin_display = null
+	var/atom/movable/screen/crew_shadekin/crew_shadekin_display = null
```
Comment (TheLordME) [inline code/modules/mob/mob_defines.dm:346]: why cant you use the shadekin_display for crewkin? Answer in 5 sentences or less

File: code/modules/species/shadekin/shadekin_abilities.dm [hunk 11]
```diff
@@ -54,8 +54,8 @@
 	if(!(ability_flags & AB_PHASE_SHIFTED))
 		log_debug("[src] attempted to shift with [watcher] visible Carbons with a  cost of [ability_cost] in a darkness level of [darkness]")
 
-	var/datum/species/shadekin/SK = species
-	if(!istype(SK))
+	//var/datum/species/shadekin/SK = species
```
Comment (TheLordME) [inline code/modules/species/shadekin/shadekin_abilities.dm:57]: this check should happen much earlier(on of the first checks

File: code/modules/species/shadekin/shadekin_blackeyed.dm [hunk 12]
```diff
@@ -107,12 +109,155 @@
...
+
 	unarmed_types = list(
 		/datum/unarmed_attack/stomp,
 		/datum/unarmed_attack/kick,
 		/datum/unarmed_attack/claws,
 		/datum/unarmed_attack/bite/sharp,
 	)
 
-/datum/species/shadekin/can_breathe_water()
+/datum/species/crew_shadekin/New()
```
Comment (TheLordME) [inline code/modules/species/shadekin/shadekin_blackeyed.dm:129]: pretty sure we dont do new() anymore, use initilize() instead

**Review by TheLordME — CHANGES_REQUESTED** [review]

File: code/modules/species/shadekin/crew_shadekin_abilities.dm (hunk 1 above)
Comment (MediHound) [reply code/modules/species/shadekin/crew_shadekin_abilities.dm:24]: Can do either. What happened was that I made the base Ability use the Species_ID, and didn't change that when I got the weaker abilities to work. Will be changed momentarily.
...
After testing it showed that I had to change it to use the "get_exact_species_id" proc.

File: code/modules/species/shadekin/crew_shadekin_abilities.dm (hunk 2 above)
Comment (MediHound) [reply code/modules/species/shadekin/crew_shadekin_abilities.dm:88]: Same as above

File: code/__DEFINES/species/species.dm (hunk 3 above)
Comment (MediHound) [reply code/__DEFINES/species/species.dm:10]: During talk with Zandario, I was told that we might as well define both of them with the new Species_ID System. This way, both Shadekin and Black-Eyed Shadekin have the same Species ID, while Black-Eyed Shadekin have the sub-species ID of Black-Eyed Shadekin.
Might be useful in the future if we have things that are supposed to work differently with Shadekin, it'll effect Black-Eyed ones as well.

File: code/modules/mob/living/carbon/human/human_powers.dm (hunk 4 above)
Comment (MediHound) [reply code/modules/mob/living/carbon/human/human_powers.dm:472]: Thanks for pointing it out, removed.

File: code/modules/mob/living/carbon/human/human_powers.dm (hunk 5 above)
Comment (MediHound) [reply code/modules/mob/living/carbon/human/human_powers.dm:424]: Went with what was there, thus I went with all caps. Will change.

File: code/modules/mob/mob_defines.dm (hunk 10 above)
Comment (MediHound) [reply code/modules/mob/mob_defines.dm:346]: I can't. Removing it from mob_defines, human_hud/human, and changing references to it in shadekin_blackeyed back to the normal shadekin_display.

File: code/modules/species/shadekin/shadekin_abilities.dm (hunk 11 above)
Comment (MediHound) [reply code/modules/species/shadekin/shadekin_abilities.dm:57]: Moved it further up. The other abilities had it pretty high up already.

File: code/modules/species/shadekin/shadekin_blackeyed.dm (hunk 12 above)
Comment (MediHound) [reply code/modules/species/shadekin/shadekin_blackeyed.dm:129]: As discussed in Discord, going to use New()

**Review by TheLordME — APPROVED** [review]

**Pull request merged**